### PR TITLE
Fix TextureRect::flip_* when used with atlas texture

### LINKED ISSUE #37526: TextureRect flip_v & flip_h gives wrong position if source is AtlasTexture
**Godot version:**
3.2 (Mono)

**Issue description:**
`TextureRect` controls have functionality to flip the chosen image in the Y or X axis using the `flip_h` or `flip_v` properties. If the texture rect's texture is coming from an `AtlasTexture` instead of an image, then calculations to render the texture are not correct.

See image below for example. The left 2 images are from an image, with `flip_v` set for the bottom row, right 2 images are from an atlas, again with 'flip_v' set on the bottom image.

AtlasFlip

You can see the content has been flipped, but is flipped outside the 'TextureRect's dimensions.

**Steps to reproduce:**
1. Create an atlas with a texture to use. The texture needs to not be on the first "row" of the atlas (i.e. at the top of the packed image. Images on the first row seem to translate correctly when flipped.
2. Create a `TextureRect` and add the image.
3. Set the `flip_v` flag. Notice the image is flipped, but moved outside the `TextureRect`

**Minimal reproduction project:**
[Testing-AtlasTextureFlip.zip](https://github.com/godotengine/godot/files/4422123/Testing-AtlasTextureFlip.zip)

### PR DESCRIPTION
**Pull request by guilhermefelipecgs**
Fix #37526

### CODE REVIEW

File: scene/gui/texture_rect.cpp [hunk 1]
```diff
@@ -95,6 +95,15 @@ void TextureRect::_notification(int p_what) {
 			} break;
 		}
 
+		Ref<AtlasTexture> p_atlas = texture;
+
+		if (p_atlas.is_valid() && region.has_no_area()) {
+			Size2 scale_size(size.width / texture->get_width(), size.height / texture->get_height());
+
+			offset.width += hflip ? p_atlas->get_margin().get_position().width * size.width / texture->get_width() * 2 : 0;
```
Comment (akien-mga) [inline scene/gui/texture_rect.cpp:103]: Didn't you mean to use `scale_size.width` here?

Comment (guilhermefelipecgs) [reply scene/gui/texture_rect.cpp:103]: Fixed.

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.2.2.

**Pull request merged**